### PR TITLE
[Refactor][Lake] Remove unused protobuf fields

### PR DESCRIPTION
**Pull request by sduzh**
This PR will break backward compatibility of Lake table, but since Lake table not released yet
it will not bring problems to our users.

### CODE REVIEW

File: gensrc/proto/lake_types.proto [hunk 1]
```diff
@@ -31,23 +23,20 @@ message TabletMetadataPB {
...
+    optional uint32 next_rowset_id = 6;
     // cumulative point rowset index
-    optional uint32 cumulative_point = 8;
+    optional uint32 cumulative_point = 7;
 }
 
 message TxnLogPB {
     message OpWrite {
         optional RowsetMetadataPB rowset = 1;
         repeated string deletes = 2;
```
Comment (wyb) [inline gensrc/proto/lake_types.proto:34]: can be removed

File: gensrc/proto/lake_types.proto [hunk 2]
```diff
@@ -31,23 +23,20 @@ message TabletMetadataPB {
     optional int64 flags = 3; 
     optional TabletSchemaPB schema = 4;
     repeated RowsetMetadataPB rowsets = 5;
-    optional PrimaryIndexMetadataPB primary_index = 6;
-    optional uint32 next_rowset_id = 7;
+    optional uint32 next_rowset_id = 6;
```
Comment (imay) [inline gensrc/proto/lake_types.proto:26]: Can't change their ordinal because of compatibility

**Review by imay — CHANGES_REQUESTED** [review]

File: gensrc/proto/lake_types.proto (hunk 2 above)
Comment (sduzh) [reply gensrc/proto/lake_types.proto]: Lake table not released yet~

**Review by wyb — DISMISSED** [review]

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Comment by wanpengfei-git** [comment]
#### [FE PR Coverage Check]


:heart_eyes: **pass** : 0 / 0 (0%)

**Review by imay — APPROVED** [review]

**Review by wyb — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Pull request merged**